File: src/Implicit/ImplicitCeiling.cs
﻿using System;

namespace TinkerWorX.AccidentalNoiseLibrary
{
    public sealed class ImplicitCeiling : ImplicitModuleBase
    {
        public ImplicitCeiling(ImplicitModuleBase source)
        {
            this.Source = source;
        }

        public ImplicitModuleBase Source { get; set; }

        public override double Get(double x, double y)
        {
            return Math.Ceiling(this.Source.Get(x, y));
        }

        public override double Get(double x, double y, double z)
        {
            return Math.Ceiling(this.Source.Get(x, y, z));
        }

        public override double Get(double x, double y, double z, double w)
        {
            return Math.Ceiling(this.Source.Get(x, y, z, w));
        }

        public override double Get(double x, double y, double z, double w, double u, double v)
        {
            return Math.Ceiling(this.Source.Get(x, y, z, w, u, v));
        }
    }
}

File: src/implicit/ImplicitCeiling.java
﻿package implicit;

import util.Maths;

public final class ImplicitCeiling extends ImplicitModuleBase {
    public ImplicitCeiling(final ImplicitModuleBase source) {
        this.Source = source;
    }

    private ImplicitModuleBase Source;

    public ImplicitModuleBase getSource() {
        return this.Source;
    }

    public void setSource(final ImplicitModuleBase source) {
        this.Source = source;
    }

    @Override
    public double get(final double x, final double y) {
        return Maths.ceil(this.Source.get(x, y));
    }

    @Override
    public double get(final double x, final double y, final double z) {
        return Maths.ceil(this.Source.get(x, y, z));
    }

    @Override
    public double get(final double x, final double y, final double z, final double w) {
        return Maths.ceil(this.Source.get(x, y, z, w));
    }

    @Override
    public double get(final double x, final double y, final double z, final double w, final double u, final double v) {
        return Maths.ceil(this.Source.get(x, y, z, w, u, v));
    }
}
